Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -860,61 +860,63 @@ class BrowserViewController: UIViewController {
         statusBarOverlay.isHidden = false
     }
 
-    /// Show the firefox home page
-    /// - Parameter inline: Inline is true when the firefox homepage is created from the tab tray, a long press on the tab bar to open a new tab or by pressing the home page button on the tab bar.
-    ///                     Inline is false when it's the zero search page, aka when the home page is shown by clicking the url bar from a loaded web page.
-    func showFirefoxHome(inline: Bool) {
-        if self.firefoxHomeViewController == nil {
-            // Firefox home page tracking i.e. being shown from awesomebar vs bottom right hamburger menu
-            let trackingValue: TelemetryWrapper.EventValue = inline ? .openHomeFromPhotonMenuButton : .openHomeFromAwesomebar
-            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .firefoxHomepage, value: trackingValue, extras: nil)
-
-            let firefoxHomeViewController = HomepageViewController(
-                profile: profile,
-                tabManager: tabManager,
-                urlBar: urlBar,
-                isZeroSearch: !inline)
-            firefoxHomeViewController.homePanelDelegate = self
-            firefoxHomeViewController.libraryPanelDelegate = self
-            firefoxHomeViewController.browserBarViewDelegate = self
-            self.firefoxHomeViewController = firefoxHomeViewController
-            addChild(firefoxHomeViewController)
-            view.addSubview(firefoxHomeViewController.view)
-            firefoxHomeViewController.didMove(toParent: self)
-            view.bringSubviewToFront(overKeyboardContainer)
+    /// Show the home page
+    /// - Parameter inline: Inline is true when the homepage is created from the tab tray, a long press
+    /// on the tab bar to open a new tab or by pressing the home page button on the tab bar. Inline is false when
+    /// it's the zero search page, aka when the home page is shown by clicking the url bar from a loaded web page.
+    func showHomepage(inline: Bool) {
+        if self.homepageViewController == nil {
+            createHomepage(inline: inline)
```

Comment:
Wondering if this is the reason we are not respecting custom home page changes. 

If we setup a custom homepage and homepage view controller is nil we create a new home page then 🤔 

_Originally posted by @nbhasin2 in https://github.com/mozilla-mobile/firefox-ios/issues/11371#issuecomment-1191891697_

Reply:
From what I get, the new tab setting isn't on the home page setting page itself. The setting you have in your screenshot is applied when we click on the home button in the toolbar.
![Screen Shot 2022-07-21 at 5 26 33 PM](https://user-images.githubusercontent.com/11338480/180318046-9ef0cba1-e626-4a40-bd18-c443e58812a3.png)

Reply:
Hmm for some reason I thought it was both. As in if a user opens a new tab then do we always show home or the tab that user has set.

It's worth checking Android and Desktop just to be in alignment.

Also its on prod as well so I don't believe it's particularly a bug on your PR